server/src/main/java/org/candlepin/model/ProductAttribute.java

```diff
@@ -163,6 +163,17 @@ public int hashCode() {
     }
 
     /**
+     * Calculates and returns a version hash for this entity. This method operates much like the
+     * hashCode method, except that it is more accurate and should have fewer collisions.
+     *
+     * @return
+     *  a version hash for this entity
+     */
+    public int getEntityVersion() {
+        return this.hashCode();
```

Comment:
Yes, I understand that, but that's the /general contract on the method definition/. The actual objects we use in our calculations have *concrete implementations* which are defined and documented or implemented ourselves.

https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#hashCode()
https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#hashCode()
https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#hashCode()
https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#hashCode()